doc/conf.py

```diff
@@ -403,6 +416,73 @@ def __call__(self, filename):
         return -version_float
 
 
+def notebook_modification_function(notebook_content, notebook_filename):
+    notebook_content_str = str(notebook_content)
+    warning_template = "\n".join(
+        [
+            "<div class='alert alert-{message_class}'>",
+            "",
+            "# JupyterLite warning",
+            "",
+            "{message}",
+            "</div>",
+        ]
+    )
+
+    message_class = "warning"
+    message = (
+        "Running the scikit-learn examples in JupyterLite is experimental and you may"
+        " encounter some unexpected behavior.\n\nThe main difference is that imports"
+        " will take a lot longer than usual, for example the first `import sklearn` can"
+        " take roughly 10-20s.\n\nIf you notice problems, feel free to open an"
+        " [issue](https://github.com/scikit-learn/scikit-learn/issues/new/choose)"
+        " about it."
+    )
+
+    markdown = warning_template.format(message_class=message_class, message=message)
+
+    dummy_notebook_content = {"cells": []}
+    add_markdown_cell(dummy_notebook_content, markdown)
+
+    code_lines = []
+
+    if "seaborn" in notebook_content_str:
+        code_lines.append("%pip install seaborn")
+    if "plotly.express" in notebook_content_str:
+        code_lines.append("%pip install plotly")
+    if "skimage" in notebook_content_str:
+        code_lines.append("%pip install scikit-image")
+    if "fetch_" in notebook_content_str:
+        code_lines.extend(
+            [
+                "%pip install pyodide-http",
+                "import pyodide_http",
+                "pyodide_http.patch_all()",
+            ]
+        )
+    # always import matplotlib and pandas to avoid Pyodide limitation with
+    # imports inside functions
+    code_lines.extend(["import matplotlib", "import pandas"])
+    # TODO temporary work-around for OpenML CORS headers issue
+    # https://github.com/openml/OpenML/issues/1135
+    if "fetch_openml" in notebook_content_str:
+        code_lines.extend(
+            [
+                "from sklearn.datasets import _openml",
+                "_openml._OPENML_PREFIX = 'https://api.openml.org/'",
```

Comment:
do we still need this?

Reply:
Yes until https://github.com/openml/OpenML/issues/1135 is fixed see https://github.com/scikit-learn/scikit-learn/pull/25887#pullrequestreview-1388679902 for more details.